### ISSUE TITLE
Port "initiateSafeTransaction" saga

### ISSUE THREAD

**Issue by ArmandoGraterol** [JoinColony/colonyCDapp#739]
Clarifications:
- This saga is quite different from the others, but basically:
  - In order to communicate with a safe that is not in our chain, we need to use an AMB (Arbitrary Message bridge).
  - For that, we first encode the data in the form of an arbitrary method call, which can also include or exclude parameters. This information, along with the target contract address (Zodiac bridge module), is passed to the AMB contracts. As soon as data is relayed from chain A to chain B by the AMB, the bridge contract on side B passes the ABI-encoded method to the target contract.
  - So basically, CDapp/Dapp > Home bridge (Gnosis) > Foreign bridge (chain of the safe) > Zodiac bridge module > Safe.
  - In order to send the message, we use the `makeArbitraryTransactions` method.
  - In the Dapp, we stored the transaction data in IPFS as it was an annotation in order to display that data where needed (action list, action, and motion page, etc). In the case of the CDapp, I'm assuming we are storing that data using our DB instead.
- This saga should handle sending a single or multiple messages.
